From 512ee1283c66ddb39b5598b2d1654568508f3fbf Mon Sep 17 00:00:00 2001
From: Sam Spilsbury <smspillaz@gmail.com>
Date: Sun, 2 Jan 2011 11:57:56 +0800
Subject: [PATCH] Use StaticGravity when sending _NET_MOVERESIZE_WINDOW requests to change vp's
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=638483
Bug-Ubuntu: http://launchpad.net/bugs/691545

ForgetGravity will cause the window manager to use the default gravity which
will shift the window by the frame extents on vp switch.
---
 libwnck/window-action-menu.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

Index: libwnck-3.2.0/libwnck/window-action-menu.c
===================================================================
--- libwnck-3.2.0.orig/libwnck/window-action-menu.c	2011-10-16 01:08:13.574231724 +0200
+++ libwnck-3.2.0/libwnck/window-action-menu.c	2011-10-16 01:08:24.514094191 +0200
@@ -231,7 +231,7 @@
           width = wnck_screen_get_width (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
           wnck_window_unstick (window);
-          wnck_window_set_geometry (window, 0,
+          wnck_window_set_geometry (window, WNCK_WINDOW_GRAVITY_STATIC,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw - width, yw,
                                     ww, hw);
@@ -252,7 +252,7 @@
           width = wnck_screen_get_width (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
           wnck_window_unstick (window);
-          wnck_window_set_geometry (window, 0,
+          wnck_window_set_geometry (window, WNCK_WINDOW_GRAVITY_STATIC,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw + width, yw,
                                     ww, hw);
@@ -273,7 +273,7 @@
           height = wnck_screen_get_height (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
           wnck_window_unstick (window);
-          wnck_window_set_geometry (window, 0,
+          wnck_window_set_geometry (window, WNCK_WINDOW_GRAVITY_STATIC,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw, yw - height,
                                     ww, hw);
@@ -294,7 +294,7 @@
           height = wnck_screen_get_height (screen);
           wnck_window_get_geometry (window, &xw, &yw, &ww, &hw);
           wnck_window_unstick (window);
-          wnck_window_set_geometry (window, 0,
+          wnck_window_set_geometry (window, WNCK_WINDOW_GRAVITY_STATIC,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw, yw + height,
                                     ww, hw);
@@ -334,7 +334,7 @@
           viewport_y = wnck_workspace_get_viewport_y (workspace);
 
           wnck_window_unstick (window);
-          wnck_window_set_geometry (window, 0,
+          wnck_window_set_geometry (window, WNCK_WINDOW_GRAVITY_STATIC,
                                     WNCK_WINDOW_CHANGE_X | WNCK_WINDOW_CHANGE_Y,
                                     xw + new_viewport_x - viewport_x,
                                     yw + new_viewport_y - viewport_y,
